ICSharpCode.CodeConverter/CSharp/CommonConversions.cs

```diff
@@ -165,7 +165,17 @@ private TypeSyntax GetFuncTypeSyntax(IMethodSymbol method)
         public TypeSyntax GetTypeSyntax(ITypeSymbol typeSymbol, bool useImplicitType = false)
         {
             if (useImplicitType || typeSymbol == null) return CreateVarTypeName();
-            return (TypeSyntax) CsSyntaxGenerator.TypeExpression(typeSymbol);
+            var vbType = SyntaxFactory.ParseTypeName(typeSymbol.ToDisplayString());
```

Comment:
I expect a VbSyntaxGenerator would do this correctly. Might be worth pulling one through and doing the logic on a generated vb tree rather than parsing typeSymbol.ToDisplayString() which might have a few unexpected quirks.

Reply:
Tried this, and it has the same problem as `CsSyntaxGenerator` - it uses the wrong casing.